### PR TITLE
ta: pkcs11: Clean up temporary_object_list in err condition

### PR DESCRIPTION
**Pull request by cmcgcc**
When handle_get() failed in create_object(), ssession obj is not removed from temporary_object_list.
Only token obj is removed inside cleanup_persistent_object().
Add LIST_REMOVE for session obj.

Signed-off-by: Mengchi Cheng <mengcc@amazon.com>

### CODE REVIEW

File: ta/pkcs11/src/object.c [hunk 1]
```diff
@@ -291,8 +291,10 @@ enum pkcs11_rc create_object(void *sess, struct obj_attrs *head,
 	handle_put(get_object_handle_db(session), obj_handle);
 	if (get_bool(head, PKCS11_CKA_TOKEN))
 		cleanup_persistent_object(obj, session->token);
-	else
+	else {
+		LIST_REMOVE(obj, link);
```
Comment (etienne-lms) [inline ta/pkcs11/src/object.c:295]: Thanks for the finding. It does the job but I think we could end up calling `LIST_REMOVE(obj, link)` only from `cleanup_volatile_obj_ref()`. It would replace this added call and those at lines 122 and 143. What do you think?

Comment (cmcgcc) [reply ta/pkcs11/src/object.c:295]: cleanup_volatile_obj_ref() is also called inside cleanup_persistent_object().  LIST_REMOVE() will be called multiple times for token obj especially in the destroy_object(). I do not think it is a good behavior although it does not create error for now. Meanwhile, obj is explicitly inserted to list inside create_object(), it is much clearer to read if it is also removed inside the same function.

Comment (cmcgcc) [reply ta/pkcs11/src/object.c:295]: Yeah, it is a good idea. Let me update the patch.

**Comment by cmcgcc** [comment]
I have updated patch. Is there any more comments?

**Comment by etienne-lms** [comment]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`
You need to fix with the commit message as per checkpatch. 76 char/line max.
I would rephrase a bit the message, my suggestion:
```
ta: pkcs11: Clean up temporary_object_list on object creation failure

Calls LIST_REMOVE() only from cleanup_volatile_obj_ref() this is always
called to finalize object release. Allocated objects are always inserted
into a list after being created and therefore need to be removed for its
owner list before the memory is released.

This changes fixes an issue when handle_get() failed in create_object()
and does not remove the reference from from temporary_object_list.
```

**Comment by jforissier** [comment]
@cmcgcc would you please add Etienne's Reviewed-by? Then I can merge this PR. Thanks!

**Comment by cmcgcc** [comment]
> @cmcgcc would you please add Etienne's Reviewed-by? Then I can merge this PR. Thanks!

Done
